profiles/common/modules/features/nexteuropa_newsroom/classes/NewsroomHelper.inc

```diff
@@ -0,0 +1,633 @@
+      $content_types = node_type_get_types();
+      // Exclude the ct used for the import.
+      unset($content_types[NEXTEUROPA_NEWSROOM_CONTENT_TYPE]);
+      unset($content_types['newsroom_selection']);
+      // Build options for selecting content types.
+      foreach ($content_types as $name => $content_type) {
+        if ($content_type->disabled !== 1) {
+          $options[$name] = $content_type->name;
+        }
+      }
+    }
+    return $options;
+  }
+
+  /**
+   * Call newsroom items import with the certain URL.
+   *
+   * @param string $url
+   *   URL to call.
+   */
+  public static function runFeedImporter($url, $importer_id) {
+    $source = feeds_source($importer_id);
+    $config = $source->getConfig();
+    $config['FeedsHTTPFetcher']['source'] = $url;
+    $source->setConfig($config);
+    $source->import();
+    // Restore the default URL in order to run importers manually or via cron.
+    $config['FeedsHTTPFetcher']['source'] = NewsroomHelper::restoreFeedImporter($importer_id);
+    $source->setConfig($config);
+    $source->save();
+  }
+
+  /**
+   * Restore default URL of importer.
+   *
+   * @param string $importer_id
+   *   Importer ID.
+   *
+   * @return string
+   *   URL.
+   */
+  public static function restoreFeedImporter($importer_id) {
+    $url = '';
+    if ($importer_id == NEXTEUROPA_NEWSROOM_TOPIC_IMPORTER || $importer_id == NEXTEUROPA_NEWSROOM_SERVICE_IMPORTER) {
+      $url = NewsroomHelper::buildImportUrl(NEXTEUROPA_NEWSROOM_TOPIC_FULL);
+    }
+
+    if ($importer_id == NEXTEUROPA_NEWSROOM_TYPE_IMPORTER) {
+      $url = NewsroomHelper::buildImportUrl(NEXTEUROPA_NEWSROOM_TYPE_FULL);
+    }
+
+    if ($importer_id == NEXTEUROPA_NEWSROOM_ITEM_IMPORTER) {
+      $url = NewsroomHelper::buildImportUrl(NEXTEUROPA_NEWSROOM_ITEM_FULL);
+    }
+
+    return $url;
+  }
+
+  /**
+   * Builds import url.
+   *
+   * @param string $url_part
+   *   Part of URL.
+   *
+   * @return string
+   *   URL.
+   */
+  public static function buildImportUrl($url_part) {
+    global $_newsroom_universe_url;
+    return $_newsroom_universe_url . $url_part;
+  }
+
+  /**
+   * Get newsroom item by Id.
+   */
+  public static function getNodeByNewsroomItemId($newsroom_id = 0) {
+    $newsroom_data = &drupal_static(__FUNCTION__);
+    $cache_key = 'newsroom_item_by_id:' . $newsroom_id;
+    if (!isset($newsroom_data)) {
+      $cache = cache_get($cache_key, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+      if ($cache) {
+        $newsroom_data = $cache->data;
+      }
+      else {
+        $query = new EntityFieldQuery();
+        $newsroom_item = $query
+          ->entityCondition('entity_type', 'node')
+          ->fieldCondition('field_newsroom_item_id', 'value', $newsroom_id, '=')
+          ->execute();
+        $newsroom_data = isset($newsroom_item['node']) ? $newsroom_item['node'] : FALSE;
+        cache_set($cache_key, $newsroom_data, NEXTEUROPA_NEWSROOM_CACHE_TABLE, time() + 3600);
```

Comment:
I have added the custom variable for every case, where use cache. So, later we can change, but I think in most cases we can use the same variable.